supplier_calendar/tests/test_supplier_calendar.py

```diff
@@ -0,0 +1,104 @@
+            'stock.stock_location_stock')
+        self.customer_location = self.env['ir.model.data'].xmlid_to_object(
+            'stock.stock_location_customers')
+        self.picking_type_out = self.env['ir.model.data'].xmlid_to_object(
+            'stock.picking_type_out')
+        self.route_buy = self.env.ref('purchase_stock.route_warehouse0_buy').id
+
+        # Create product
+        self.product = self.env['product.product'].create({
+            'name': 'test product',
+            'default_code': 'PRD',
+            'type': 'product',
+            'route_ids': [(4, self.ref('stock.route_warehouse0_mto')),
+                          (4, self.ref('purchase_stock.route_warehouse0_buy'))]
+        })
+
+        # Partner and Supplierinfo
+        self.company_partner.write({
+            'delay_calendar_type': 'supplier_calendar',
+            'factory_calendar_id': self.calendar.id
+
+        })
+        self.seller_01 = self.supplier_info.create({
+            'name': self.company_partner.id,
+            "product_id": self.product.id,
+            'product_tmpl_id': self.product.product_tmpl_id.id,
+            'delay': 3
+        })
+
+    def test_01_purchase_order_with_supplier_calendar(self):
+        # Create a customer picking
+        customer_picking = self.env['stock.picking'].create({
+            'location_id': self.stock_location.id,
+            'location_dest_id': self.customer_location.id,
+            'partner_id': self.company_partner.id,
+            'picking_type_id': self.picking_type_out.id,
+        })
+
+        customer_move = self.env['stock.move'].create({
+            'name': 'move out',
+            'location_id': self.stock_location.id,
+            'location_dest_id': self.customer_location.id,
+            'product_id': self.product.id,
+            'product_uom': self.product.uom_id.id,
+            'product_uom_qty': 80.0,
+            'procure_method': 'make_to_order',
+            'picking_id': customer_picking.id,
+            'date_expected': '2097-01-14 09:00:00'  # Monday
+        })
+
+        customer_move._action_confirm()
+
+        purchase_order = self.env['purchase.order'].search(
+            [('partner_id', '=', self.company_partner.id)])
+        self.assertTrue(purchase_order, 'No purchase order created.')
+        date_order = fields.Date.to_date(purchase_order.date_order)
+        wednesday = fields.Date.to_date('2097-01-09 09:00:00')
+        self.assertEqual(date_order, wednesday)  # Wednesday
+
+    def test_02_purchase_order_supplier_calendar_global_leaves(self):
+        # Global leaves
+        self.calendar.write({
+            'global_leave_ids': [
+                (0, False, {
+                    'name': 'Test',
+                    'date_from': '2097-01-14',  # Monday
+                    'date_to': '2097-01-19',  # Saturday
+                }),
+            ],
+        })
+
+        reference = "2097-01-14 09:00:00"  # Monday
```

Comment:
I was not asking to remove the comment, I was wondering wether it should be like this or not, kind of opening the discussion :thinking: